### PR TITLE
ENH: Adds support for altitude_binning across more cosmic files

### PR DESCRIPTION
**Pull request by rstoneback**
Implemented a new form of binning with the change to xarray. The native groupby in xarray doesn't work like I need along multiple dimensions. Thus, I implemented something new. Started from a stack overflow [answer](https://stackoverflow.com/questions/38013778/is-there-any-numpy-group-by-function) and went from there as needed. 

Doesn't work for scintillation files since they lack the `MSL_alt` variable.

### CODE REVIEW

File: pysatCDAAC/instruments/cosmic_gps.py [hunk 1]
```diff
@@ -406,104 +407,117 @@ def load(fnames, tag=None, inst_id=None, altitude_bin=None):
...
+            # Get all variables/coordinates using same dimensions as MSL_alt
+            # and sort.
+            msl_dims = output['MSL_alt'].dims
+
+            var_labels = [var for var in output.data_vars
+                          if output[var].dims == msl_dims]
+            coord_labels = [coord for coord in output.coords
+                            if output[coord].dims == msl_dims]
+
+            all_labels =[]
```
Comment (rstoneback) [inline pysatCDAAC/instruments/cosmic_gps.py:448]:
```suggestion
            all_labels = []
```

File: pysatCDAAC/instruments/cosmic_gps.py [hunk 2]
```diff
@@ -406,104 +407,117 @@ def load(fnames, tag=None, inst_id=None, altitude_bin=None):
...
-                output = new_set
         elif tag == 'atmprf':
             # Set up coordinates.
             coord_labels = ['MSL_alt', 'Lat', 'Lon', 'Azim']
 
-            # Apply coordinates to loaded data.
-            output = output.set_coords(coord_labels)
         elif tag == 'sonprf':
             # Set up coordinates.
             coord_labels = ['MSL_alt', 'lat', 'lon']
```
Comment (jklenzing) [inline pysatCDAAC/instruments/cosmic_gps.py:417]: not in supported tags, not run according to coveralls

File: pysatCDAAC/tests/test_instruments.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def teardown_class(self):
     def test_altitude_bin_keyword(self, inst_dict):
         """Test altitude bin keyword."""
 
-        if inst_dict['tag'] != 'ionprf':
+        if inst_dict['tag'] == 'scnlv1':
             assert True
```
Comment (jklenzing) [inline pysatCDAAC/tests/test_instruments.py:68]: can use pytest
```suggestion
            pytest.skip("Binning not available for scnlv1 data")
```

File: pysatCDAAC/instruments/cosmic_gps.py [hunk 4]
```diff
@@ -315,7 +315,7 @@ def load(fnames, tag=None, inst_id=None, altitude_bin=None):
         satellite id or None (default=None)
     altitude_bin : integer
         Number of kilometers to bin altitude profiles by when loading.
-        Currently only supported for tag='ionprf'.
+        Works for all files except `tag=scnlv1`.
```
Comment (jklenzing) [inline pysatCDAAC/instruments/cosmic_gps.py:318]:
```suggestion
        Works for all files except `tag='scnlv1'`.
```

File: pysatCDAAC/instruments/cosmic_gps.py [hunk 5]
```diff
@@ -32,7 +32,7 @@
     None supported
 altitude_bin
     Number of kilometers to bin altitude profiles by when loading.
-    Currently only supported for tag='ionprf'.
+    Works for all files except `tag=scnlv1`.
```
Comment (jklenzing) [inline pysatCDAAC/instruments/cosmic_gps.py:35]:
```suggestion
    Works for all files except `tag='scnlv1'`.
```

**Comment by jklenzing** [comment]
We might want to split the 2013 reprocess out into its own product (inst_id?) again.  Looking at some of the other data products at CDAAC, having the preferential download might overcomplicate things from a dev standpoint.

File: pysatCDAAC/instruments/cosmic_gps.py [hunk 6]
```diff
@@ -408,104 +411,143 @@ def load(fnames, tag=None, inst_id=None, altitude_bin=None):
...
+
+        elif tag == 'ionphs':
+            # Set up coordinates.
+            coord_labels = ['caL1Snr', 'pL1Snr', 'pL2Snr',
+                            'xLeo', 'yLeo', 'zLeo', 'xdLeo', 'ydLeo', 'zdLeo',
+                            'xGps', 'yGps', 'zGps', 'xdGps', 'ydGps', 'zdGps',
+                            'exL1', 'exL2']
+
+        elif tag == 'podtec':
+            # Set up coordinates.
```
Comment (aburrell) [inline pysatCDAAC/instruments/cosmic_gps.py:444]: Remove duplicate comments or expand to be unique.

**Review by jklenzing — APPROVED** [review]
Thanks @rstoneback!

**Comment by rstoneback** [comment]
Thanks @jklenzing.

**Pull request merged**